jme3-core/src/main/java/com/jme3/material/MaterialDef.java

```diff
@@ -195,4 +200,85 @@ public void addTechniqueDef(TechniqueDef technique) {
         return techniques.keySet();
     }
 
+    @Override
+    public MaterialDef jmeClone() {
+        try {
+            return (MaterialDef) super.clone();
+        } catch (final CloneNotSupportedException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public void cloneFields(final Cloner cloner, final Object original) {
+
+        final Map<String, MatParam> oldMatParams = matParams;
+
+        matParams = new HashMap<>();
+
+        for (final Map.Entry<String, MatParam> entry : oldMatParams.entrySet()) {
+            matParams.put(entry.getKey(), cloner.clone(entry.getValue()));
+        }
+
+        final Map<String, List<TechniqueDef>> oldTechniques = this.techniques;
+
+        techniques = new HashMap<>();
+
+        for (final Map.Entry<String, List<TechniqueDef>> entry : oldTechniques.entrySet()) {
+            techniques.put(entry.getKey(), cloner.clone(entry.getValue()));
+        }
+    }
+
+    @Override
+    public void write(final JmeExporter ex) throws IOException {
```

Comment:
No. Matdef should be stored as j3md.

Reply:
you are right for most cases, but not for all cases :(

Reply:
I am right in every case. It's been already discussed in the core team. Binary serialization should contain a scene. By doing this you open the special case where you could save only a materialDef in a j3o, which makes no sense and is very confusing for users. Save your materialDefs as j3md. There is even a serializer that does this in core. Look for J3mdExporter.

Reply:
@Nehon I don't talk about serialization j3md to j3o ;) you can use binary exporter to serialize jme objects to other places.

Reply:
Then this has nothing to do in the engine.

Reply:
You often tend to change things in the engine that are only very particular uses cases that wouldn't be needed in a regular game. This increase a lot the clutter of code in the engine and is wrong in many ways. If you have special use case work around them in your own projects.

Reply:
@Nehon I see only one work around, I need to copy-paste all material's classes :(

Reply:
Well then you have to think more about how you could redesign your editor so that you don't need to do that... I really don't get what you are doing.

Reply:
@Nehon I just want to have binary serialization for j3md, I didn't think that is a big problem for you, now I see that you have the strong position about it.